airbyte-cdk/python/airbyte_cdk/sources/file_based/error_handlers/file_based_discover_error_handler.py

```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+import logging
+from typing import Optional
+
+from airbyte_cdk.sources.file_based.exceptions import ConfigValidationError, InvalidSchemaError, SchemaInferenceError
+from airbyte_cdk.sources.utils.discover_error_handler import AbstractDiscoverErrorHandler
+
+
+class FileBasedDiscoverErrorHandler(AbstractDiscoverErrorHandler):
+    def handle_discover_error(self, logger: logging.Logger, exception: Exception) -> Optional[Exception]:
+        """
+        Default File Based source implementation of the discover error handler. Logs the error or raises the exception.
+        """
+        exceptions_to_log = (InvalidSchemaError, SchemaInferenceError, ConfigValidationError)
+
+        if isinstance(exception, exceptions_to_log):
+            logger.error(
+                f"Error occurred while discovering stream and therefore stream will not be added to the configured catalog: {exception}",
+                exc_info=True,
+            )
+            return None
+        else:
+            return exception
```

Comment:
could we raise the exception instead of returning it [just to task the source to raise it](https://github.com/airbytehq/airbyte/pull/44798/files#diff-51c861828e0a614ef6ee21390f3ba17d7138549f0575051fdb24868ee05eabbbR87-R88)?

Reply:
I think so. We can implement another type of error handler if we want different handling